index.html

```diff
@@ -614,6 +614,97 @@ <h4>
         </section>
       </section>
     </section>
+    <section>
+      <h2>
+        Automation
+      </h2>
+      <p>
+        The Device Posture API pose a challenge to test authors, as fully
+        exercising interface requires physical hardware devices. To address this
+        challenge this document defines a [[WEBDRIVER2]] [=extension commands=]
+        that allows users to control the reported device posture and simulate a
+        real device.
+      </p>
+      <h3>
+        Extension Commands
+      </h3>
+      <h4>
+        Set device posture
+      </h4>
+      <table class="simple">
+        <tr>
+          <th>
+            HTTP Method
+          </th>
+          <th>
+            [=extension command URI Template|URI Template=]
+          </th>
+        </tr>
+        <tr>
+          <td>
+            POST
+          </td>
+          <td>
+            /session/{session id}/deviceposture
+          </td>
+        </tr>
+      </table>
+      <p>
+        This [=extension commands|extension command=] changes device posture to a specific
+        {{DevicePostureType}}.
+      </p>
+      <table class="simple">
+        <caption>
+          Properties of the parameters argument used by this algorithm
+        </caption>
+        <tr>
+          <th>
+            Parameter name
+          </th>
+          <th>
+            Value type
+          </th>
+          <th>
+          Required
+          </th>
+        </tr>
+        <tr>
+          <td>
+            posture
+          </td>
+          <td>
+            String
+          </td>
+          <td>
+            yes
+          </td>
+        </tr>
+      </table>
+      <p>
+        The [=remote end steps=] are:
+      </p>
+      <ol class="algorithm" data-cite="WEBDRIVER2">
+        <li>Let |posture| be the result of invoking
+        <a data-cite="!WEBDRIVER2#dfn-getting-properties">get a property</a>
+        "posture" from |parameters|.
+        </li>
+        <li>If |posture| is not a [=string=], return [=error=] with
+        [=error code|WebDriver error code=] [=invalid argument=].
+        </li>
+        <li>If |posture| is neither "{{DevicePostureType/continuous}}" nor
+        "{{DevicePostureType/folded}}", return [=error=] with
+        [=error code|WebDriver error code=] [=invalid argument=].
+        </li>
+        <li>Let |document| be the [=top-level traversable=]'s
```

Comment:
We're almost there, but now the question is where is your top-level traversable coming from :-) You're referring to a general concept in the HTML spec, not something that was derived from something else (which would be the case if you had referred to https://html.spec.whatwg.org/multipage/document-sequences.html#nav-top instead, for example).

As I said, look at the Generic Sensors spec, it needed to deal with this very same problem.

Reply:
Please clarify. This is one example of Generic sensor spec:
"Let topLevelVirtualSensorMapping be the [current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context)'s [top-level traversable](https://html.spec.whatwg.org/multipage/document-sequences.html#bc-traversable)'s [virtual sensor mapping](https://www.w3.org/TR/generic-sensor/#virtual-sensor-mapping)."

And this is device-posture:
"Let document be the [top-level traversable](https://html.spec.whatwg.org/multipage/document-sequences.html#top-level-traversable)'s [active document](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-document)."

Is the difference "[current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context)'s"?  So it should say:
"Let document be the [current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context)'s [top-level traversable](https://html.spec.whatwg.org/multipage/document-sequences.html#top-level-traversable)'s [active document](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-document)."?

Reply:
> Is the difference "[current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context)'s"? So it should say:
> "Let document be the [current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context)'s [top-level traversable](https://html.spec.whatwg.org/multipage/document-sequences.html#top-level-traversable)'s [active document](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-document)."?

Yes (as explained in our call), but not exactly: your internal slot should be tied to a top-level traversable, not a Document.

Reply:
Fixed https://github.com/JuhaVainio/W3C-device-posture/pull/1/commits/a9d3d5a7bfc454eb439350719c23b4a0c1935c74